### ISSUE TITLE
[Bug]: Cannot open file named "index.html"

### ISSUE THREAD

**Issue by wojciu35** [nextcloud/server#31068]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I uploaded a file called "index.html" (just working on a site) to a folder in my Nextcloud account using the Windows sync client. The file is shown in the Web GUI, however when I try to download it, i get "404 not found", when I try to open the file details it returns "Error while loading the file data"
It happens no matter which folder I'm in, and by using "create text document" function I get error "Could not fetch file details "index.html""
Such file cannot be deleted, the only way I could get rid of it was to delete it from the server's file system and then use occ files:scan.

### Steps to reproduce

1. upload or create a file called "index.html"

### Expected behavior

the file should behave like any others

### Installation method

Manual installation

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database engine version

MariaDB

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.9.0
  - activity: 2.15.0
  - admin_audit: 1.13.0
  - announcementcenter: 6.1.1
  - audioplayer: 3.2.4
  - breezedark: 23.2.0
  - calendar: 3.0.5
  - circles: 23.0.0
  - cloud_federation_api: 1.6.0
  - comments: 1.13.0
  - contacts: 4.0.7
  - contactsinteraction: 1.4.0
  - dashboard: 7.3.0
  - dav: 1.21.0
  - deck: 1.6.0
  - federatedfilesharing: 1.13.0
  - federation: 1.13.0
  - files: 1.18.0
  - files_automatedtagging: 1.13.0
  - files_external: 1.15.0
  - files_pdfviewer: 2.4.0
  - files_rightclick: 1.2.0
  - files_sharing: 1.15.0
  - files_trashbin: 1.13.0
  - files_versions: 1.16.0
  - files_videoplayer: 1.12.0
  - firstrunwizard: 2.12.0
  - logreader: 2.8.0
  - lookup_server_connector: 1.11.0
  - mail: 1.11.6
  - news: 17.0.1
  - nextcloud_announcements: 1.12.0
  - notes: 4.3.0
  - notifications: 2.11.1
  - oauth2: 1.11.0
  - password_policy: 1.13.0
  - photos: 1.5.0
  - privacy: 1.7.0
  - provisioning_api: 1.13.0
  - recommendations: 1.2.0
  - serverinfo: 1.13.0
  - settings: 1.5.0
  - sharebymail: 1.13.0
  - side_menu: 2.3.3
  - spreed: 13.0.3
  - support: 1.6.0
  - survey_client: 1.11.0
  - systemtags: 1.13.0
  - tasks: 0.14.2
  - terms_of_service: 1.9.2
  - text: 3.4.0
  - theming: 1.14.0
  - twofactor_backupcodes: 1.12.0
  - twofactor_totp: 6.2.0
  - updatenotification: 1.13.0
  - user_ldap: 1.13.0
  - user_status: 1.3.1
  - viewer: 1.7.0
  - weather_status: 1.3.0
  - welcome: 1.0.1
  - workflow_pdf_converter: 1.8.0
  - workflowengine: 2.5.0
Disabled:
  - apporder
  - browser_warning
  - bruteforcesettings
  - camerarawpreviews
  - encryption
  - files_fulltextsearch
  - files_fulltextsearch_tesseract
  - fulltextsearch
  - passwords
  - polls
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Additional info

i am using nginx reverse proxy to access nextcloud with no prior problems

**Comment by szaimen** [comment]
Thank you for taking the time to report a bug! 👍

As this seems to be a setup issue I would like to ask you to raise your question in the forums: https://help.nextcloud.com 
If you wish support with setup issues from Nextcloud GmbH we offer this as part of the Nextcloud subscription. Learn more about this at https://nextcloud.com/enterprise/

**Comment by wojciu35** [comment]
Hello,
very sorry to bother you again.
I opened an issue on https://help.nextcloud.com/t/cant-access-file-named-index-html/133405/3
They say it's a normal behaviour for html files, however I am at a loss since I can't work on a project.
A friend with a sync client 3.4.1 seemingly successfully uploaded a folder that contained and index.html file, but when he copied the folder to another location it returned an error "File was removed from the server". It appeared on my PC, but trying to open it i get "0x80070185: the cloud operation was unsuccessful". The file is present and fine in the server's file system.

I really don't know what to do with this and since the logs do not show anything I can't even troubleshoot it.
It appeared quite recently, as this folder contained an html file for a long time, was synced and edited across many devices. It started to cause trouble after I repaired the mtime which was broken after updating the sync client to 3.4.0
Can my issue be caused by the sync client? I am on 3.4.2, my server is on 23.0.0
